src/jit/lsra.cpp

```diff
@@ -9884,9 +9884,18 @@ void LinearScan::resolveEdge(BasicBlock*      fromBlock,
         (resolveType == ResolveSharedCritical) ? REG_NA : getTempRegForResolution(fromBlock, toBlock, TYP_INT);
 #endif // !_TARGET_XARCH_
     regNumber tempRegFlt = REG_NA;
+#ifdef _TARGET_ARM_
+    // ARM32 requires double temp register in addition to interger and float temp registers.
+    regNumber tempRegDbl = REG_NA;
+#endif
     if ((compiler->compFloatingPointUsed) && (resolveType != ResolveSharedCritical))
     {
         tempRegFlt = getTempRegForResolution(fromBlock, toBlock, TYP_FLOAT);
+
+#ifdef _TARGET_ARM_
+        // ARM32 requires double temp register in addition to interger and float temp registers.
+        tempRegDbl = getTempRegForResolution(fromBlock, toBlock, TYP_DOUBLE);
```

Comment:
> The resolution uses only one register at a time, so it would be better to reserve only a TYP_DOUBLE register on arm, and then its lower half can be used when resolving a TYP_FLOAT.

When considering code structure, I also think the suggested approach is better.
However I'm afraid it may affect code quality sometimes when we need `TYP_FLOAT` temp but there  there is a free `TYP_FLOAT` register and no free `TYP_DOUBLE` register. I've looked into `getTempRegForResolution()` and it does not guarantee reserving a temp register.
What do you think of it?